### ISSUE TITLE
Xml.serialize stripComments parameter doesn't work

### ISSUE THREAD

**Issue by olafos** [scala/scala-xml#183]
stripComments parameter in Utility.serialize is ignored due to unfortunate pattern matching expression:
```
      x match {
        case c: Comment if !stripComments => c buildString sb
        case s: SpecialNode               => s buildString sb
      ...
```
since Comment is also a SpecialNode the second pattern is applied after the first one doesn't match.

**Comment by ashawley** [comment]
Indeed.  This defect was introduced in 7662284a on Mar-08-2012, which references scala/bug#1118.  That bug was about empty XML elements. So it seems there was some incidental refactoring done while fixing the bug that introduced the bug or it may have been reverting 8f433bc6, that had it right, but I can't tell.

**Comment by ashawley** [comment]
Fixed in #184
